src/Symfony/Component/HttpClient/UriTemplateHttpClient.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\Service\ResetInterface;
+
+class UriTemplateHttpClient implements HttpClientInterface, ResetInterface
+{
+    use DecoratorTrait;
+
+    /**
+     * @param (\Closure(string $url, array $vars): string)|null $expander
+     */
+    public function __construct(HttpClientInterface $client = null, private ?\Closure $expander = null, private array $defaultVars = [])
+    {
+        $this->client = $client ?? HttpClient::create();
+    }
+
+    public function request(string $method, string $url, array $options = []): ResponseInterface
+    {
+        $vars = [...$this->defaultVars, ...$options['vars'] ?? []];
```

Comment:
`if (isset($options['vars']))` around to do as little as possible when vars are not used